gui/ros_gui/gui/configpanel/configpanel.py

```diff
@@ -0,0 +1,357 @@
+            param_name (str): Name of the parameter.
+        """
+        result = await local_file_picker('~/.astazero/ATOS', multiple=False, show_hidden_files=True)
+        ui.notify(f'You selected {result}')
+        if not result:
+            return
+        self.parameters[param_name] = str(result)
+        self.set_parameter(node_name, param_name, str(result))
+
+    async def pick_files(self, node_name, param_name) -> None:
+        """ Opens a file picker and sets the selected files as the parameter value.
+
+        Args:
+            node_name (str): Name of the node that the parameter belongs to.
+            param_name (str): Name of the parameter.
+        """
+        result = await local_file_picker('~/.astazero/ATOS', multiple=True, show_hidden_files=True)
+        ui.notify(f'You selected {result}')
+        if not result:
+            return
+        self.parameters[param_name] = str(result)
+        self.set_parameter(node_name, param_name, str(result))
+
+    def set_param_callback(self, future, param_name) -> None:
+        """ Callback function for the set parameters service call. Notifies the user if the call was successful or not.
+        
+        Args:
+            future (Future): Future object returned by the set parameters service call (Bool).
+            param_name (str): Name of the parameter.
+        """
+        try:
+            response = future.result()
+        except Exception as e:
+            self.get_logger().info('Service call failed %r' % (e,))
+            return
+        self.get_logger().debug(f'Setting parameter {param_name} was {"successful" if response.result.successful else "unsuccesful"}')
+        with self.splitter:
+            ui.notify(f'Setting parameter {param_name} was {"successful" if response.result.successful else "unsuccesful"}')
+
+    def set_parameter(self, node_name, param_name, param_value) -> None:
+        """ Sets the value for a specific parameter.
+
+        Args:
+            node_name (str): Name of the node that the parameter belongs to.
+            param_name (str): Name of the parameter.
+            param_value (any): Value of the parameter.
+        """
+        self.get_logger().debug(f'Setting parameter {param_name} in {node_name} to {param_value}')
+        ui.notify(f'Setting parameter {param_name} in {node_name} to {param_value}')
+
+        service_timeout_counter = 0
+        client = self.client_list[node_name]["set_params_client"]
+        while not client.wait_for_service(timeout_sec=1.0):
+            service_timeout_counter += 1
+            self.get_logger().debug('service not available, waiting again...')
+            if service_timeout_counter >= MAX_TIMEOUT:
+                ui.notify(f'Service not available after {MAX_TIMEOUT} seconds, please try again later')
+                self.get_logger().info(f'Service not available after {MAX_TIMEOUT} seconds, please try again later')
+                return
```

Comment:
Since we do wait_for_service in multiple places we should extract it to reduce code duplication.